hibernate-core/src/test/java/org/hibernate/test/bytecode/enhancement/lazy/proxy/MappedSuperclassWithEmbeddableTest.java

```diff
@@ -0,0 +1,148 @@
+import javax.persistence.Embedded;
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.MappedSuperclass;
+
+import org.hibernate.boot.registry.StandardServiceRegistryBuilder;
+import org.hibernate.cfg.AvailableSettings;
+
+import org.hibernate.testing.TestForIssue;
+import org.hibernate.testing.bytecode.enhancement.BytecodeEnhancerRunner;
+import org.hibernate.testing.bytecode.enhancement.EnhancementOptions;
+import org.hibernate.testing.junit4.BaseNonConfigCoreFunctionalTestCase;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static org.hamcrest.core.IsNull.notNullValue;
+import static org.hibernate.testing.transaction.TransactionUtil.doInHibernate;
+import static org.junit.Assert.assertThat;
+
+/**
+ * @author Andrea Boriero
+ */
+@RunWith(BytecodeEnhancerRunner.class)
+@EnhancementOptions(lazyLoading = true, inlineDirtyChecking = true)
+public class MappedSuperclassWithEmbeddableTest extends BaseNonConfigCoreFunctionalTestCase {
+
+	@Override
+	protected void configureStandardServiceRegistryBuilder(StandardServiceRegistryBuilder ssrb) {
+		super.configureStandardServiceRegistryBuilder( ssrb );
+		ssrb.applySetting( AvailableSettings.ALLOW_ENHANCEMENT_AS_PROXY, "true" );
+		ssrb.applySetting( AvailableSettings.FORMAT_SQL, "false" );
+		ssrb.applySetting( AvailableSettings.USE_SECOND_LEVEL_CACHE, "false" );
+		ssrb.applySetting( AvailableSettings.GENERATE_STATISTICS, "true" );
+	}
+
+	@Override
+	public Class<?>[] getAnnotatedClasses() {
+		return new Class<?>[] { TestEntity.class };
+	}
+
+	@Before
+	public void prepare() {
+		doInHibernate( this::sessionFactory, s -> {
+			TestEntity testEntity = new TestEntity( "2", "test" );
+			s.persist( testEntity );
+		} );
+	}
+
+	@After
+	public void tearDown() {
+		doInHibernate( this::sessionFactory, s -> {
+			s.createQuery( "delete from TestEntity" ).executeUpdate();
+		} );
+	}
+
+	@Test
+	public void testIt() {
+		doInHibernate( this::sessionFactory, s -> {
+			TestEntity testEntity = s.get( TestEntity.class, "2" );
+			assertThat( testEntity, notNullValue() );
+		} );
+	}
+
+	@MappedSuperclass
+	public static abstract class BaseEntity {
+		@Embedded
+		private EmbeddedValue superField;
+
+		public EmbeddedValue getSuperField() {
+			return superField;
+		}
+
+		public void setSuperField(EmbeddedValue superField) {
+			this.superField = superField;
+		}
+	}
+
+	@Embeddable
+	public static class EmbeddedValue implements Serializable {
+		@Column(name = "super_field")
+		private String superField;
+
+		public EmbeddedValue() {
+		}
+
+		private EmbeddedValue(String superField) {
+			this.superField = superField;
+		}
+
+		@Override
+		public boolean equals(Object o) {
+			if ( this == o ) {
+				return true;
+			}
+			if ( o == null || getClass() != o.getClass() ) {
+				return false;
+			}
+			EmbeddedValue that = (EmbeddedValue) o;
+			return superField.equals( that.superField );
```

Comment:
I think `Objects.equals(superField, that.superField)` would be more robust and align well with the `hashCode` implementation, though it would be no concern if we are sure `superField` is never null.

Reply:
This is a test case :)
We know it's not going to be null